### PR TITLE
Access Refactoring

### PR DESCRIPTION
**Pull request by phsmit**
This PR builds on #900 . It is not a complete product yet, but it shows what kind of idea I have with the access refactoring.

In this first commit it already removes a great number lines of code and further refactorings might get it down even more. 

Things that still need to be done:
 - Migration needs to be written
 - Testing all changes to see if nothing breaks.

### CODE REVIEW

**Comment by Unknwon** [comment]
Since changes are a lot, would you like we create a `access_refactor` branch on main repository to test it?

**Comment by phsmit** [comment]
If a branch works good for you, it is good for me. I can PR both collaboration and access_refactor to it

File: models/access.go [hunk 1]
```diff
@@ -4,77 +4,151 @@
...
+type AccessMode int
 
 const (
-	READABLE AccessType = iota + 1
-	WRITABLE
+	NoAccess AccessMode = iota
+	ReadAccess
+	WriteAccess
+	AdminAccess
+	OwnerAccess
```
Comment (Unknwon) [inline models/access.go]: I would prefer

```go
ACCESS_MODE_NONE
ACCESS_MODE_READ
ACCESS_MODE_WRITE
ACCESS_MODE_ADMIN
ACCESS_MODE_OWNER
```

**Comment by phsmit** [comment]
Superseded by #926

**Pull request closed without merging**